libs/entitlement/src/test/java/org/elasticsearch/entitlement/runtime/policy/PolicyManagerTests.java

```diff
@@ -229,15 +264,32 @@ public void testRequestingModuleWithStackWalk() throws IOException, ClassNotFoun
         );
     }
 
+    public void testAgentsEntitlements() throws IOException, ClassNotFoundException {
+        var agentsClass = makeClassInItsOwnModule();
+        var notAgentsClass = makeClassInItsOwnModule();
+        var policyManager = new PolicyManager(
+            createEmptyTestServerPolicy(),
+            List.of(new CreateClassLoaderEntitlement()),
+            Map.of(),
+            c -> "test",
+            agentsClass.getModule(),
+            NO_ENTITLEMENTS_MODULE
+        );
+        ModuleEntitlements agentsEntitlements = policyManager.getEntitlements(agentsClass);
+        assertThat(agentsEntitlements.hasEntitlement(CreateClassLoaderEntitlement.class), is(true));
+        ModuleEntitlements notAgentsEntitlements = policyManager.getEntitlements(notAgentsClass);
+        assertThat(notAgentsEntitlements.hasEntitlement(CreateClassLoaderEntitlement.class), is(false));
```

Comment:
Nit: I'd like a check that uses two unnamed modules, to show that using named/unnamed is no longer a factor.
It should be simple, create 2 classloaders, load 2 classes (even the same) with them, and use classloader.getUnnamedModule() to pass to PolicyManager fot the agent module for one of the 2.
Even if it's the "same" class, one should pass, the other should not.

Reply:
That's a good idea! I will add this.